JsonSchema.Generation/SchemaGeneratorConfiguration.cs

```diff
@@ -34,11 +35,42 @@ public class SchemaGeneratorConfiguration
 	/// <remarks>
 	/// This can be replaced with any `Func&lt;string, string&gt;`.
 	/// </remarks>
-	public PropertyNamingMethod PropertyNamingMethod
+	[Obsolete($"Use {nameof(PropertyNameResolver)} instead.")]
+	public PropertyNamingMethod? PropertyNamingMethod
 	{
-		get => _propertyNamingMethod ??= PropertyNamingMethods.AsDeclared;
+		get => _propertyNamingMethod;
 		set => _propertyNamingMethod = value;
 	}
+
+	/// <summary>
+	/// Gets or sets the property name resolving method. Default is <see cref="PropertyNameResolvers.ByJsonPropertyName"/>.
+	/// </summary>
+	/// <remarks>
+	/// This can be replaced with any `Func&lt;MemberInfo, string&gt;`.
+	/// </remarks>
+	public PropertyNameResolver PropertyNameResolver
+	{
+		get
```

Comment:
You might not be 100% happy with the complex getter. Cause the rest of the class is just a relatively simple Poco.

But, the logic for the forward/backward compat needed to go somewhere, and all of the if-return can be removed, when the old prop will be removed.